### PR TITLE
Add rekey attribute to inde-sdk WalletCredentials type

### PR DESCRIPTION
**Pull request by MosCD3**
Signed-off-by: Mostafa <mostafa.youssef@ontario.ca>

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [<<url here>>](https://github.com/hyperledger/aries-framework-javascript/issues/671)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## indy-sdk ([<kbd>unpkg</kbd>](https://unpkg.com/browse/indy-sdk@latest/))
was missing the following properties:
1. capi
1. setRuntimeConfig
1. setDefaultLogger
1. setLogger
1. issuerRotateCredentialDefStart

<details>
<summary>as well as these 75 other properties...</summary>
<p>issuerRotateCredentialDefApply, proverGetCredentials, proverSearchCredentials, proverFetchCredentials, proverCloseCredentialsSearch, proverDeleteCredential, updateRevocationState, toUnqualified, setKeyMetadata, getKeyMetadata, cryptoAuthCrypt, cryptoAuthDecrypt, cryptoAnonDecrypt, replaceKeysStart, replaceKeysApply, storeTheirDid, keyForDid, setEndpointForDid, getEndpointForDid, setDidMetadata, getDidMetadata, getMyDidWithMeta, listMyDidsWithMeta, qualifyDid, submitAction, multiSignRequest, buildGetDdoRequest, buildAttribRequest, buildNodeRequest, buildGetValidatorInfoRequest, buildGetTxnRequest, buildPoolConfigRequest, buildPoolRestartRequest, buildPoolUpgradeRequest, buildAuthRuleRequest, buildAuthRulesRequest, buildGetAuthRuleRequest, buildTxnAuthorAgreementRequest, buildDisableAllTxnAuthorAgreementsRequest, buildAcceptanceMechanismsRequest, appendRequestEndorser, getResponseMetadata, deleteWalletRecordTags, getSchema, getCredDef, purgeSchemaCache, purgeCredDefCache, isPairwiseExists, createPairwise, listPairwise, getPairwise, setPairwiseMetadata, createPaymentAddress, listPaymentAddresses, addRequestFees, parseResponseWithFees, buildGetPaymentSourcesRequest, buildGetPaymentSourcesWithFromRequest, parseGetPaymentSourcesResponse, parseGetPaymentSourcesWithFromResponse, buildPaymentReq, parsePaymentResponse, preparePaymentExtraWithAcceptanceData, buildMintReq, buildSetTxnFeesReq, buildGetTxnFeesReq, parseGetTxnFeesResponse, buildVerifyPaymentReq, parseVerifyPaymentResponse, getRequestInfo, signWithAddress, verifyWithAddress, refreshPoolLedger, listPools, generateWalletKey</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e5700c70f356e266fd9589a6a012f90918039e80
</p>

**Review by TimoGlastra — CHANGES_REQUESTED** [review]
Could you add a test @MosCD3? We can't merge this ourselves without it

File: types/indy-sdk/index.d.ts [hunk 1]
```diff
@@ -304,6 +304,7 @@ export interface WalletStorageConfig {
 
 export interface WalletCredentials {
     key: string;
+    rekey?: string;
```
[the remark this answers is not in the document]
Comment (MosCD3) [reply types/indy-sdk/index.d.ts:307]: Its already here

export interface OpenWalletCredentials extends WalletCredentials {
    rekey_derivation_method?: KeyDerivationMethod | undefined;
}

The other thing that I can do here is 

export interface OpenWalletCredentials extends WalletCredentials {
    rekey?: string;
    rekey_derivation_method?: KeyDerivationMethod | undefined;
}

and remove it from 'WalletCredentials' 
bottom line the 'rekey' has to be there or I will not be able to pass it to 'openWallet' method in AFJ

[the remark this answers is not in the document]
Comment (TimoGlastra) [reply types/indy-sdk/index.d.ts:307]: A great, missed that. 

`rekey` is only allowed when opening the wallet right? In that case I'd put it on the `OpenWalletCredentials` interface

[the remark this answers is not in the document]
Comment (MosCD3) [reply types/indy-sdk/index.d.ts]: Done

[the remark this answers is not in the document]
Comment (TimoGlastra) [reply types/indy-sdk/index.d.ts]: Thanks. Could you also add a test? I can't merge without

**Comment by MosCD3** [comment]
> 

I don't want mess things up, could we go for that now and later enhance it with tests? Just to get things going

File: types/indy-sdk/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MosCD3) [reply types/indy-sdk/index.d.ts]: I managed to add it as far as I understood the way you add tests here, let me know if I messed up

File: types/indy-sdk/indy-sdk-tests.ts [hunk 2]
```diff
@@ -349,7 +354,8 @@ indy.createRevocationState(
 // indy.generateWalletKey({})
 // indy.exportWallet(handle, exportConfig)
 // indy.importWallet(walletConfig, walletCredentials, exportConfig)
-// indy.listPools()
+// indy.import { id } from '../apicache/index.d';
```
Comment (TimoGlastra) [inline types/indy-sdk/indy-sdk-tests.ts:357]: Can you remove this?

**Comment by MosCD3** [comment]
@TimoGlastra  is there any thing missing here? if not could please merge so that I continue on the AFJ ticket

**Review by TimoGlastra — APPROVED** [review]

**Comment by TimoGlastra** [comment]
Ready to merge

**Comment by TimoGlastra** [comment]
Ready to merge

**Pull request merged**